front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx

```diff
@@ -169,47 +205,26 @@ const RollingStockEditorParameterFormColumn = ({
                 id={property.title}
                 inputDataTestId={`${property.title}-input`}
                 label={t(property.title)}
-                typeValue={property.type}
-                type={optionValue}
-                value={
-                  !isNil(rollingStockValues[property.title])
-                    ? (rollingStockValues[property.title] as string | number)
-                    : ''
-                }
+                currentValue={{
+                  unit: currentParam.unit,
+                  value: currentParam.value,
+                }}
                 options={property.units.map((unit) => ({
-                  id: `${property.title}-${unit}`,
+                  id: unit,
                   label: unit,
                 }))}
-                handleType={(type) => {
-                  setRollingStockValues({
-                    ...rollingStockValues,
-                    [property.title]: type.value !== '' ? Number(type.value) : undefined,
-                  });
-                  setOptionValue(type.type as string);
-                }}
-                min={property.min}
-                max={property.max}
+                handleUnit={(option) => handleUnitChange(option, property)}
+                min={currentParam.min}
+                max={currentParam.max}
                 isInvalid={
-                  rollingStockValues[property.title]?.toString().length === 0 ||
-                  (rollingStockValues[property.title] as number) < (property.min as number) ||
-                  (rollingStockValues[property.title] as number) > (property.max as number)
-                }
-                errorMsg={
-                  property.max
-                    ? t('errorMessages.minMaxRangeError', {
-                        min: property.min?.toString().replace('.', ','),
-                        max: property.max?.toString().replace('.', ','),
-                      })
-                    : t('errorMessages.minRangeError', {
-                        min: property.min?.toString().replace('.', ','),
-                      })
+                  currentParam.value === undefined ||
+                  currentParam.value < currentParam.min ||
+                  currentParam.value > currentParam.max
                 }
-                step="any"
-                sm
-                condensed
-                orientation="right"
-                textRight
-                disableUnitSelector
+                errorMsg={t('errorMessages.minMaxRangeError', {
+                  min: currentParam.min.toString().replace('.', ','),
+                  max: floor(currentParam.max, 6).toString().replace('.', ','),
```

Comment:
Why do you floor ? Maybe create a utils function ? 
```suggestion
                  max: currentParam.max.toString().replace('.', ','),
```

Reply:
It was to avoid this

![Capture d’écran 2024-03-07 à 13 05 09](https://github.com/osrd-project/osrd/assets/113104137/4e640125-503f-4df5-9615-747b4a2dafa2)